### PR TITLE
✨ [Part1] Add golangci lint yaml

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
This PR adds golangci.yml to the repo which enables linters to help improve the codebase overall.
The `golangci.YAML` has been mostly copied from the Cluster-api repo with changes adapted for this repo. 
All issues in the codebase that came up after adding the linter file have been fixed

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]
Nice!

File: baremetal/utils.go [hunk 1]
```diff
@@ -73,7 +73,9 @@ func patchIfFound(ctx context.Context, helper *patch.Helper, host client.Object)
 	err := helper.Patch(ctx, host)
 	if err != nil {
 		notFound := true
-		if aggr, ok := err.(kerrors.Aggregate); ok {
+		var aggr kerrors.Aggregate
+		if ok := errors.As(err, &aggr); ok {
+			// if aggr, ok := err.(kerrors.Aggregate); ok {
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply baremetal/utils.go]: Thanks removed!

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

**Review by fmuyassarov — APPROVED** [review]

**Pull request merged**